homeassistant/components/otbr/__init__.py

```diff
@@ -18,13 +20,24 @@ class OTBRData:
     """Container for OTBR data."""
 
     url: str
+    api: python_otbr_api.OTBR
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the Open Thread Border Router component."""
+    websocket_api.async_setup(hass)
+    return True
 
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up an Open Thread Border Router config entry."""
+    api = python_otbr_api.OTBR(entry.data["url"], async_get_clientsession(hass), 10)
+    try:
+        await api.get_active_dataset_tlvs()
+    except python_otbr_api.OTBRError as exc:
+        raise ConfigEntryNotReady from exc
```

Comment:
Is this is to wait for the addon to startup? Why is that needed though?
Do we need to do anything about outage of the REST API for other reasons though, such as OTBR not managed by our addon, or the addon not running?

Reply:
We always test if services are available when we set them up, so users can know if their system is bad.

Reply:
Raising `ConfigEntryNotReady` does an exponential backoff of retry times.
In the typical case, an integration needs access to the external service to set itself up. In this case the integration only proxies calls for its users, which means the users need to wait also for the exponential backoff to cool down, I don't see how that's helpful.

Reply:
removed it for now. You can add it back when we evolve the integration